sdk/evm/tbrv3/contract.ts

```diff
@@ -118,27 +133,79 @@ export interface Transfer {
 
 export class Tbrv3 {
 
-  readonly address: string;
-
-  constructor(public readonly provider: ethers.Provider, public readonly network: NetworkMain, address: string = Tbrv3.addresses[network]) {
+  constructor(
+    public readonly provider: ethers.Provider,
+    public readonly network: NetworkMain,
+    public readonly address: string = Tbrv3.addresses[network],
+    private readonly gasToken: string,
+  ) {
     this.address = address;
+
+  }
+
+  static connect(
+    provider: ethers.Provider,
+    network: NetworkMain,
+    chain: Chain,
+    gasToken?: string,
+    address?: string,
+  ) {
+    if (address === undefined && this.addresses[network] === "TBD") {
+      throw new Error(`Tbrv3 address needs to be provided for network ${network}`);
+    }
+
+    const defaultGasToken = this.gasTokens[network][chain];
+    if (gasToken === undefined) {
+      if (defaultGasToken === undefined) {
+        throw new Error(`Gas token address needs to be provided for network ${network} and chain ${chain}`);
+      }
+      gasToken = defaultGasToken;
+    } else if (defaultGasToken !== undefined && gasToken.toLowerCase() !== defaultGasToken?.toLowerCase()) {
+      throw new Error(`Unexpected gas token address ${gasToken} for network ${network} and chain ${chain}`);
+    }
+
+    return new Tbrv3(provider, network, address, gasToken);
+  }
+
+  /**
+   * @deprecated This method will be removed soon. The sdk won't use ethers directly anymore.
+   */
+  static fromRpcUrlStatic(rpc: string, network: NetworkMain, chain: Chain, gasToken?: string, address?: string): Tbrv3 {
+    return this.connect(new ethers.JsonRpcProvider(rpc, undefined, {staticNetwork: true}), network, chain, gasToken, address);
+  }
+
+  /**
+   * @deprecated This method will be removed soon. The sdk won't use ethers directly anymore.
+   */
+  static fromRpcUrl(rpc: string, network: NetworkMain, chain: Chain, gasToken?: string, address?: string): Tbrv3 {
+    return this.connect(new ethers.JsonRpcProvider(rpc), network, chain, gasToken, address);
   }
 
   static readonly addresses: Record<NetworkMain, string> = {
     Mainnet: "TBD",
     Testnet: "TBD",
   }
```

Comment:
Should go into the SDKv2, should be a constMap.

Reply:
This one? I'm not sure it should :thinking:
Do we put all addresses separate from their protocol bindings?
But I do thing the gas tokens should go into the sdkv2.

Reply:
> I'm not sure it should 🤔 Do we put all addresses separate from their protocol bindings?
Not all, but the ones that are relevant for token bridge transfers. [TBRv2 addresses are in the SDKv2 too](https://github.com/wormhole-foundation/wormhole-sdk-ts/blob/main/core/base/src/constants/contracts/tokenBridgeRelayer.ts).

Reply:
I see. We can definitely do a PR that merges this SDK in there. It's going to make awkward doing deployments though :thinking: 

I think we should just merge upstream the stable API and treat this as a package for internal use during deployments mostly.